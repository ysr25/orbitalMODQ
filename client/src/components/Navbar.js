import React, { Component } from "react";
import { Link } from "react-router-dom";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import logo from "../logo.jpg";

export default class CustomNavbar extends Component {
  logout = (e) => {
    e.preventDefault();

    this.props
      .api("post", "/users/logout")
      .then((res) => this.props.updateUser())
      .catch((err) => console.log("Logout error"));
  };

  render() {
    const loggedIn = this.props.loggedIn;

    return (
      <Navbar bg="light" expand="sm" sticky="top">
        <Link to="/" className="navbar-brand" onClick={this.props.updateUser}>
          <img src={logo} width="50" height="50" alt="MODQ Logo" />
          MODQ
        </Link>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse>
          <Nav className="mr-auto">
            <Link to="/" className="nav-link" onClick={this.props.updateUser}>
              Home
            </Link>
            <Link
              to="/reviews/new"
              className="nav-link"
              onClick={this.props.updateUser}
            >
              Create a Post
            </Link>
            {loggedIn ? (
              <>
                <Link
                  to="/users/edit/"
                  className="nav-link"
                  onClick={this.props.updateUser}
                >
                  Edit Account Details
                </Link>
                <Link to="#" className="nav-link" onClick={this.logout}>
                  Log Out
                </Link>
              </>
            ) : (
              <>
                <Link
                  to="/users/signup"
                  className="nav-link"
                  onClick={this.props.updateUser}
                >
                  Register
                </Link>
                <Link
                  to="/users/login"
                  className="nav-link"
                  onClick={this.props.updateUser}
                >
                  Log In
                </Link>
              </>
            )}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}
