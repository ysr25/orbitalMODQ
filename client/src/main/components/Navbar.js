import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import '../App.css'

class Navbar extends Component {
  logout = (e) => {
    e.preventDefault()

    this.props.api('post', '/users/logout')
      .then(res => this.props.updateUser())
      .catch(err => {
        console.log('Logout error')
      })
  }

  render () {
    const loggedIn = this.props.loggedIn

    return (
      <div className='collpase navbar-collapse'>
        <ul className='navbar-nav mr-auto'>
          <li className='navbar-item'>
            <Link to='/' className='nav-link' onClick={this.props.updateUser}>
              Home
            </Link>
          </li>
          <li className='navbar-item'>
            <Link to='/modreviews/newpost' className='nav-link' onClick={this.props.updateUser}>
              Create A Post
            </Link>
          </li>
          {loggedIn ? (
            <li className='navbar-item'>
              <Link to='/users/edit/' className='nav-link' onClick={this.props.updateUser}>
                Edit Account Details
              </Link>
            </li>
          ) : (
            <li className='navbar-item'>
              <Link to='/users/signup' className='nav-link' onClick={this.props.updateUser}>
                Register
              </Link>
            </li>
          )}
          {loggedIn ? (
            <li className='navbar-item'>
              <Link to='#' className='nav-link' onClick={this.logout}>
                Log Out
              </Link>
            </li>
          ) : (
            <li className='navbar-item'>
              <Link to='/users/login' className='nav-link' onClick={this.props.updateUser}>
                Log In
              </Link>
            </li>
          )}
        </ul>
      </div>
    )
  }
}

export default Navbar
