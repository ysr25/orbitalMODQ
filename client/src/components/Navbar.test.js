import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Navbar from './Navbar'

const mockUpdateUser = jest.fn(() => false)

it('renders navigation bar (logged in)', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar loggedIn updateUser={mockUpdateUser} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders navigation bar (not logged in)', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar loggedIn={false} updateUser={mockUpdateUser} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
