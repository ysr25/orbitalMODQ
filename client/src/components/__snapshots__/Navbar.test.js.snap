// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navigation bar (logged in) 1`] = `
<div
  className="collpase navbar-collapse"
>
  <ul
    className="navbar-nav mr-auto"
  >
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/modreviews/newpost"
        onClick={[Function]}
      >
        Create A Post
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/users/edit/"
        onClick={[Function]}
      >
        Edit Account Details
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders navigation bar (not logged in) 1`] = `
<div
  className="collpase navbar-collapse"
>
  <ul
    className="navbar-nav mr-auto"
  >
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/modreviews/newpost"
        onClick={[Function]}
      >
        Create A Post
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/users/signup"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="nav-link"
        href="/users/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </li>
  </ul>
</div>
`;
